app/portainer/home/EnvironmentList/KubeconfigButton.tsx

```diff
@@ -34,48 +81,154 @@ export function KubeconfigButton({ environments }: Props) {
       category: 'kubernetes',
     });
 
-    showKubeconfigModal(environments);
+    selection.clear();
+    setShowDialog(true);
+  }
+
+  function handleClose() {
+    setShowDialog(false);
+  }
+
+  function handleDownload() {
+    confirmKubeconfigSelection();
+  }
+
+  function isKubeconfigButtonVisible(environments: Environment[]) {
+    if (window.location.protocol !== 'https:') {
+      return false;
+    }
+    return environments.some((env) => isKubernetesEnvironment(env.Type));
+  }
+
+  function useKubeServiceExpiry() {
+    const { isSuccess, isError, data, error } = useQuery('', async () => {
+      const expiryMessage = await kcService.expiryMessage();
+      return expiryMessage;
+    });
+    if (isSuccess) {
+      return data;
+    }
+    if (isError) {
```

Comment:
if there's an error it will show this message every rerender. use `onError` in the useQuery options, or use meta. see https://github.com/portainer/portainer-ee/blob/4ce0b7d541b0fcc1d8d5a09aa716d243261c70f1/app/portainer/settings/queries.ts#L12